lib/tokenlist.cpp

```diff
@@ -1768,6 +1768,10 @@ void TokenList::validateAst() const
             tok = tok->link();
             continue;
         }
+        if (Token::simpleMatch(tok, ". decltype (")) { // skip trailing return type
```

Comment:
We should have a valid AST in a trailing decltype. It doesn't seem good to skip this.

Reply:
We don't have that at the moment:
~~~
namespace N {
	struct S { using T = int; };
}
void f() {	
	auto g = []() -> N::S::T {
		return {};
	};
};
~~~
~~~
##AST
[bar.cpp:5]
=
|-g
`-[ f:00000299479A0050
  `-(
    `-{ 'int'

[bar.cpp:6]
return
`-{
~~~
How would the TRT connect to the AST anyway?

Reply:
But this doesn't skip the validation for this case. If we are going to skip trailing return types then we should probably skip all not just ones that start with a decltype.